test/test_data/test_target_validator.py

```diff
@@ -176,66 +156,87 @@ def test_targetvalidator_supported_types_noclassification(input_data_targettest)
         'numpy_multiclass',
         'list_binary',
         'list_multiclass',
-        'sparse_bsr_nonan',
-        'sparse_coo_nonan',
-        'sparse_csc_nonan',
-        'sparse_csr_nonan',
-        'sparse_lil_nonan',
         'openml_2',
     ),
     indirect=True
 )
 def test_targetvalidator_supported_types_classification(input_data_targettest):
+    y = input_data_targettest  # Just to remove visual clutter
+
     validator = TargetValidator(is_classification=True)
-    validator.fit(input_data_targettest)
-    transformed_y = validator.transform(input_data_targettest)
-    if sparse.issparse(input_data_targettest):
-        assert sparse.issparse(transformed_y)
-    else:
-        assert isinstance(transformed_y, np.ndarray)
-    epected_shape = np.shape(input_data_targettest)
-    if len(epected_shape) > 1 and epected_shape[1] == 1:
-        # The target should have (N,) dimensionality instead of (N, 1)
-        epected_shape = (epected_shape[0], )
-    assert epected_shape == np.shape(transformed_y)
-    assert np.issubdtype(transformed_y.dtype, np.number)
+    validator.fit(y)
+
+    y_encoded: np.ndarray = validator.transform(y)
+    y_inverse: np.ndarray = validator.inverse_transform(y_encoded)
+
+    # Assert the validator was fitted
     assert validator._is_fitted
 
-    # Because there is no classification, we do not expect a encoder
-    if not sparse.issparse(input_data_targettest):
-        assert validator.encoder is not None
+    # Assert it returns y_inverse, y_encoded as ndarray
+    assert isinstance(y_encoded, np.ndarray)
+    assert isinstance(y_inverse, np.ndarray)
```

Comment:
Is this what you meant by "not preserving the type"?

Reply:
Yes, of the three components of the input:
* `type(x)` is not inverted (e.g. `DataFrame`)
* `x.dtype` is inverted where possible (e.g. `np.float32`)
* encoded columns of x are inverted (e.g. OrdinalEncoding)